question17/main.js

```diff
@@ -0,0 +1,151 @@
+} 
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch('https://mocki.io/v1/09593e78-80ba-4765-940e-3c70fb46078c'));
+  }, 3000);
+})
+
+//Get json data
+async function callApi() {
+  try {
+    const res = await getData;
+    if (!res.ok) {
+      throw new Error(`サーバーリクエストに失敗しました: ${res.status}`);
+    }
+    const json = await res.json();
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+async function init() {
+  const imgData = await callApi();
+  if (imgData){
+    renderImgUiElement(imgData);
+    } else {
+    imgListsWrapper.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+
+function renderImgUiElement(imgData) {
+  renderNextBtn(imgData);
+  renderPrevBtn(imgData);
+  renderPageNumElement(imgData);
+  imgLists.appendChild(getImgListsFragment(imgData));
+}
+
+const renderPageNumElement = (imgData) => {
+  const numberOfPage = createElementWithClassName("p", "pageNumber");
+  imgListsWrapper.appendChild(numberOfPage);
+  numberOfPage.id = "js-number";
+  numberOfPage.textContent = `1 / ${imgData.length}`;  
+}
+
+//get Img Elements 
+const getImgListsFragment = (imgData) => {
+  const fragmentImglists = document.createDocumentFragment();
+  for (let i = 0; i < imgData.length; i++) {
+    const imgList = createElementWithClassName("li", "imgList");
+    const img = createElementWithClassName("img", `img_0${[i + 1]}`);
+    img.src = imgData[i].img;
+    img.alt = imgData[i].description;
+    i === 0 && imgList.classList.add("is-show");
+    fragmentImglists.appendChild(imgList).appendChild(img);
+  }
+  return fragmentImglists;
+};
+
+let imgNum = 0;
+const isShowElement = document.getElementsByClassName("is-show");
+const imgList = document.getElementsByClassName("imgList");
+
+
+const renderPrevBtn = (imgData) => {
+  const prevBtn = createElementWithClassName("button", "prev");
+  prevBtn.id = 'js-prevbtn';
+  prevBtn.textContent = '◀︎';
+  imgListsWrapper.appendChild(prevBtn);
+  prevBtn.disabled = true;
+
+  prevBtn.addEventListener("click", function() {
+    imgNum -= 1;
+    if (imgNum < 0) {
+      imgNum = 0;
+    }
+    isShowElement[0].classList.remove('is-show'); 
+    imgList[imgNum].classList.add('is-show');
+    switchDisableForBtn(imgNum,imgData); 
+    setNumberOfPage(imgNum,imgData); 
+  });
+}
+
+const renderNextBtn = (imgData) => {
+  const nxtBtn = createElementWithClassName("button", "next");
+  nxtBtn.id = 'js-nextbtn';
+  nxtBtn.textContent = '▶︎';
+  imgListsWrapper.appendChild(nxtBtn);
+
+  nxtBtn.addEventListener("click", function() {
+    imgNum += 1;
+    if (imgNum >= imgData.length) {
+      imgNum = imgData.length - 1;
+    }
+    isShowElement[0].classList.remove('is-show'); 
+    imgList[imgNum].classList.add('is-show');
+    switchDisableForBtn(imgNum,imgData);
+    setNumberOfPage(imgNum,imgData); 
```

Comment:
IMO
Since imgNum is defined globally, I think it can be referenced without passing it as an argument.

Reply:
Thank you for the review! Yes it is true..! I was terribly mistaken.:fearful: The argument here has been removed.
Please confirm.
fixed commit: https://github.com/csakyo/homework/pull/31/commits/168ddf101913cacc55f69068ae021a54b159fd0f